backend/inspirehep/records/api/mixins.py

```diff
@@ -313,3 +317,43 @@ def is_bucket_uuid(test_str):
     @staticmethod
     def is_hash(test_str):
         return test_str and len(test_str) == 40
+
+
+class ConferencePaperAndProccedingsMixin:
+    def clean_conference_literature_relation(self):
+        ConferenceLiterature.query.filter_by(literature_uuid=self.id).delete()
+
+    def create_conferences_relations(self, document_type):
+        conferences_pids = self.get_linked_pids_from_field(
+            "publication_info.conference_record"
+        )
+        conferences = self.get_records_by_pids(conferences_pids)
+        for conference in conferences:
+            if conference.get("deleted") is not True:
+                db.session.add(
+                    ConferenceLiterature(
+                        conference_uuid=conference.id,
+                        literature_uuid=self.id,
+                        relationship_type=ConferenceToLiteratureRelationshipType(
+                            document_type
+                        ),
+                    )
+                )
+
+    def update_conference_paper_and_proccedings(self):
+        self.clean_conference_literature_relation()
+        document_type = self.get("document_type")[0]
```

Comment:
this is wrong, it doesn't have to be the first in the list, you should check if any is `proceedings` or `conference paper`.